bundles/org.openhab.binding.verisure/src/main/resources/ESH-INF/thing/thing-types.xml

```diff
@@ -0,0 +1,594 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<thing:thing-descriptions bindingId="verisure"
+	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xmlns:thing="https://openhab.org/schemas/thing-description/v1.0.0"
+	xsi:schemaLocation="https://openhab.org/schemas/thing-description/v1.0.0 https://openhab.org/schemas/thing-description-1.0.0.xsd">
+
+	<bridge-type id="bridge">
+		<label>Verisure Bridge</label>
+		<description>This bridge represents the http://mypages.verisure.com web page connector.</description>
+
+		<channels>
+			<channel id="status" typeId="status" />
+		</channels>
+
+		<properties>
+			<property name="vendor">Verisure</property>
+		</properties>
+
+		<config-description>
+			<parameter name="username" type="text" required="true">
+				<label>Username</label>
+				<description>The username used to login to http://mypage.verisure.com.</description>
+			</parameter>
+			<parameter name="password" type="text" required="true">
+				<label>Password</label>
+				<context>password</context>
+				<description>The password used to login to http://mypage.verisure.com.</description>
+			</parameter>
+			<parameter name="refresh" type="integer">
+				<label>Refresh Interval</label>
+				<description>Specifies the refresh interval in seconds.</description>
+				<default>600</default>
+			</parameter>
+			<parameter name="pin" type="text">
+				<label>Pin Code</label>
+				<description>The username's pin code to arm/disarm alarm and lock/unlock door. Comma separated in case of many installations.</description>
+				<context>password</context>
+			</parameter>
+		</config-description>
+
+	</bridge-type>
+
+	<thing-type id="alarm">
+		<supported-bridge-type-refs>
+			<bridge-type-ref id="bridge" />
+		</supported-bridge-type-refs>
+
+		<label>Verisure Alarm</label>
+		<description>A Verisure Alarm.</description>
+
+		<channels>
+			<channel id="numericStatus" typeId="numericStatus" />
+			<channel id="alarmStatus" typeId="alarmStatus" />
+			<channel id="changedByUser" typeId="changedByUser" />
+			<channel id="changedVia" typeId="changedVia" />
+			<channel id="timestamp" typeId="timestamp" />
+			<channel id="installationName" typeId="installationName" />
+			<channel id="installationId" typeId="installationId" />
+			<channel id="setAlarmStatus" typeId="setAlarmStatus" />
+		</channels>
+
+		<properties>
+			<property name="vendor">Verisure</property>
+		</properties>
+
+		<config-description>
+			<parameter name="deviceId" type="text" required="true">
+				<label>Device Id</label>
+				<description>Unique Id.</description>
+			</parameter>
+		</config-description>
+
+	</thing-type>
+
+	<thing-type id="smartLock">
+		<supported-bridge-type-refs>
+			<bridge-type-ref id="bridge" />
+		</supported-bridge-type-refs>
+
+		<label>Yale Doorman SmartLock</label>
+		<description>Integrated Yale Doorman SmartLock.</description>
+
+		<channels>
```

Comment:
Is there specific reason why there are dedicated channels for current states and another channels for change purposes (e.g. smartLockStatus/setSmartLockStatus)? I think this violates the openHAB princible.

Reply:
The set-channel is defined as a switch and is used to change state and the other channel is defined as a String that displays a textual representation of the state (LOCKED/UNLOCKED) that can be shown in  a sitemap.

You mean only have a a channel defined as a switch?

Reply:
> You mean only have a a channel defined as a switch?
Yes. That's why openHAB have command and state separation. Commands will try to change the item/channel state and state itself represents the actual state at the moment. If user want to open the lock, commands from UI or rules will be send to the channel, which tries to open the lock. If lock is successfully opened, lock state will change and binding will update channel state accordingly. Then UI's (sitemaps) and rules can be used in "standard" way.

Item have label defination and state transformations, which can be used to translate item state to human readable format (if state itself is not clear enough). Also sitemaps have mappings function to translate items state to human readable format. With these functions, user can choose the representation format and language.

String channel/item is intended to be used for information which varies, like currently played song name or radio/tv channel name.

Reply:
OK, then I'll keep the Switch channel but rename it to `smartLockStatus` and remove the String channel. Same for `smartPlug`.
Alarm status is represented by a Number, since you can have 3 different states. I'll then keep the Number channel but also rename that to `alarmStatus` and remove the old String channel `alarmStatus`.

Reply:
Fixed and pushed!